deeplearning4j-keras/src/test/java/org/deeplearning4j/keras/DeepLearning4jEntryPointTest.java

```diff
@@ -1,31 +1,45 @@
 package org.deeplearning4j.keras;
 
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
 import org.apache.commons.io.FileUtils;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
 
+import javax.annotation.Nullable;
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import static org.deeplearning4j.keras.StringsEndsWithPredicate.endsWith;
 
 public class DeepLearning4jEntryPointTest {
 
     private DeepLearning4jEntryPoint deepLearning4jEntryPoint = new DeepLearning4jEntryPoint();
 
-    public @Rule ExpectedException thrown = ExpectedException.none();
+    public
+    @Rule
+    ExpectedException thrown = ExpectedException.none();
 
     @Test
     public void shouldFitTheSampleSequentialModel() throws Exception {
         // Given
-        File model = prepareResource("theano_mnist/model.h5");
-        File features = prepareResource("theano_mnist/x_train.h5");
-        File labels = prepareResource("theano_mnist/y_train.h5");
+        final File model = prepareResource("theano_mnist/model.h5");
+        final Path features = prepareFeatures("theano_mnist");
+        final Path labels = prepareLabels("theano_mnist");
 
         EntryPointFitParameters entryPointParameters = EntryPointFitParameters
                 .builder()
                 .modelFilePath(model.getAbsolutePath())
-                .trainFeaturesFile(features.getAbsolutePath())
-                .trainLabelsFile(labels.getAbsolutePath())
+                .trainFeaturesDirectory(features.toAbsolutePath().toString())
+                .trainLabelsDirectory(labels.toAbsolutePath().toString())
                 .batchSize(128)
```

Comment:
Better. Why not also allow files though? The internal implementation of this would just be equivalent to calling director on the file's parent.